compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -471,11 +476,12 @@ public BSymbol resolvePrefixSymbol(SymbolEnv env, Name pkgAlias, Name compUnit)
             BSymbol symbol = entry.symbol;
             long tag = symbol.tag;
 
-            if ((tag & SymTag.XMLNS) == SymTag.XMLNS) {
+            if ((tag & SymTag.XMLNS) == SymTag.XMLNS &&
+                    ((BXMLNSSymbol) symbol).compUnit.equals(compUnit)) {
                 return symbol;
             }
 
-            if ((tag & SymTag.IMPORT) == SymTag.IMPORT &&
+            if (!((tag & SymTag.XMLNS) == SymTag.XMLNS) && (tag & SymTag.IMPORT) == SymTag.IMPORT &&
```

Comment:
Do we need `!((tag & SymTag.XMLNS) == SymTag.XMLNS) ` this check here?

Reply:
In SymTag class we have the following tags for import and xmlns,
```java
public static final long IMPORT = 1;
public static final long XMLNS = 1 << 13 | IMPORT;
```

Thus 
* XMLNS_TAG = 8193
* IMPORT_TAG = 1

Consider a case where `tag` variable is XMLNS tag (8193).

Here 
`tag & SymTag.IMPORT = 8193 & 1 = 1 = SymTag.IMPORT`
Thus when XMLNS is the tag variable doing AND operation with IMPORT tag is making it equal to IMPORT tag.
Thus we first need to ensure that it is not a XMLNS tag.
Otherwise BPackageSymbol cast is going to fail.